apis/core/v1beta1/openstackcontrolplane_types.go

```diff
@@ -168,6 +181,47 @@ type OpenStackControlPlaneSpec struct {
 	ExtraMounts []OpenStackExtraVolMounts `json:"extraMounts,omitempty"`
 }
 
+// TLSSection defines the desired state of TLS configuration
+type TLSSection struct {
+	// +kubebuilder:validation:Optional
+	// +operator-sdk:csv:customresourcedefinitions:type=spec
+	// PublicEndpoints tls configuration
+	PublicEndpoints TLSPublicEndpointSection `json:"publicEndpoints,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	// +operator-sdk:csv:customresourcedefinitions:type=spec
+	// InternalEndpoints tls configuration
+	InternalEndpoints TLSInternalEndpointSection `json:"internalEndpoints,omitempty"`
+
+	// +kubebuilder:validation:optional
+	//+operator-sdk:csv:customresourcedefinitions:type=spec
+	// Secret containing any additional CA certificates, which should be added to deployment pods
+	tls.Ca `json:",inline"`
+}
+
+// TLSPublicEndpointSection defines the desired state of public TLSEndpoint configuration
+type TLSPublicEndpointSection struct {
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default=true
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,xDescriptors={"urn:alm:descriptor:com.tectonic.ui:booleanSwitch"}
+	// Enabled - Whether TLS should be enabled for endpoint type
+	Enabled bool `json:"enabled"`
+
+	// +kubebuilder:validation:Optional
+	// +operator-sdk:csv:customresourcedefinitions:type=spec
+	// Issuer - cert-manager issuer to be used for the endpoint type. If not specified a self signed will be created.
+	Issuer *string `json:"issuer,omitempty"`
```

Comment:
@olliewalsh @vakwetu @Deydra71 should we change this to be just a secret name the user should provide, instead of an issuer to be used? I think thats what we discussed in one of our meetings.

Reply:
as discussed with @olliewalsh @Deydra71 on slack,  I'll change the PR to not have a named issuer as parameter and instead a named secret which must contain public/private key

Reply:
this is now changed in the latest version. Per service in the apiOverride you can specify in the `apiOverride.tls.secretName` the secret which must hold the tls.key, tls.crt and ca.crt . if the secret does not exist, or does not have those entries in the data, the operator logs an error.

```
  keystone:
    apiOverride:
      route: {}
      tls:
        secretName:
```